### DISCUSSION TITLE
71

### DISCUSSION THREAD

**Discussion by jojoldu** [jojoldu/blog-comments#736] [Comments]
http://jojoldu.tistory.com/71

**Comment by jojoldu** [comment]
작성자: seungdols

안녕하세요. AOP에 관한 글 잘 읽고 있습니다. 그런데, 의문점이 하나 드는 것은 관점을 분리하여 중복된 코드를 떼어놓음과 자주 사용하는 기능 코드를 분리해낸다라는 점은 매우 우수하다고 판단되지만, AOP를 사용함에 따라 코드 분석이나 이해도는 크게 떨어 지지 않을까요 ? 
저는 아직 코드로 보면 막상 이해가 쉽지 않더라구요...;;;;

**Comment by russell-g5628** [comment]
감사합니다. 잘보고 갑니다.

**Comment by chan920520** [comment]
안녕하세요~ Spring을 공부하다가 AOP와 Transaction에 공부해야되서 찾다가 좋은 글 보고 갑니다. 

혹시 실례가 되지 않는다면 기본적으로 위와 같은 개념이 잘 잡혀있는 책 또는 소스를 알려주실수 있으신가요??
답변주시면 감사하겠습니다.

**Comment by jojoldu** [comment]
@bibis0520 
안녕하세요 :)
일반적으로 AOP를 가장 잘 공부하기 위해서는 토비의 스프링책이 좋습니다.
완전 밑바닥 개념을 다 짚고가서 개인적으로 제대로 이해하기 좋았습니다.
내용이 어려우시다면 최범균님의 스프링책도 추천드립니다.
다만 최범균님의 책은 예제 중심이라 원하시는것과 다를수도 있습니다.

이외에는 인프런에 올라온 [백기선님의 강좌](https://www.inflearn.com/course/spring-framework_core/)를 추천합니다.

**Comment by chan920520** [comment]
토비의 스프링에 대해서 많이 들어봤는데 한번 봐야될거같네요
단순히 소스를 작성하는게 아니라 이젠 내부적으로 어떤 과정이 존재하는지 많이 궁굼하더라구요.

늦은시간인데 답변 바로 주셔서 감사합니다.
추천해주신 자료들 참고할께요!
수고하세요!!!

2019년 1월 25일 (금) 21:49, 이동욱 <notifications@github.com>님이 작성:

**Comment by jar100** [comment]
안녕하세요 중요한 질문은 아닌데 요즘 공부하면서 @Bean 과 @Component 에 대해 고민을 많이하고 있습니다. aop를 적용하실 때 따로 빈을 등록해주셨는데

@Component
@Aspect
public class Performance {
}

이런식으로 선언해 줘도 되더라구요

저는 @Component 로 하면 따로 빈을 등록해주게 되는 번거로움이 없으져 더 낫다고 생각되는데
 
주인장님께서는 어느 방식이 낫다고 생각하시나요

**Comment by G-ONL** [comment]
안녕하세요 출간하신 책도 잘 읽고 평소에 블로그도 잘 보고 있습니다.

이번에 AOP를 공부하면서 약간 용어에 대해 헷갈려서 여쭤봅니다.

언급해주신 

"제일 먼저 보이는 @Around는 어드바이스입니다."

@Around , @Before, @After, @AfterReturning, @AfterThrowing 이렇게 5가지의 시점이 존재하잖아요.

이것들이 JoinPoint에 해당하는게 아닌가 싶어서요.

Advice는 말 그대로 수행될 로직이고, 저 5가지 Annotation들은 의미가 Target의 메서드를 언제 합류해서 사용하라는 의미이면 JoinPoint가 맞지 않을까 싶어서요!

우문일 수 있지만 현답을 주시리라 믿고 질문해 봅니다.

제가 참고한 자료는 밑에 자료 입니다.
https://www.inflearn.com/questions/29032
https://www.inflearn.com/questions/21173

**Comment by dhmin5693** [comment]
자바지기님 블로그가 죽었는지 접속하면 이상한 광고 페이지가 뜹니다.
수정해주셔야 할 것 같아서 댓글 남기고 가요~
